File: hackerrank/Day8/main.py
if __name__ == '__main__':
    n = int(input())

    phone_book = {}

    for _ in range(n):
        entry = input().split()
        name = entry[0]
        number = entry[1]
        phone_book[name] = number

    while True:
        try:
            query = input()
        except EOFError:
            break

        if query in phone_book:
            print(f"{query}={phone_book[query]}")
        else:
            print("Not found")

File: hackerrank/Day8/main.cs
using System;
using System.Collections.Generic;
using System.IO;
class Solution {
    static void Main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution */
int n = Convert.ToInt32(Console.ReadLine().Trim());

        Dictionary<string, string> phoneBook = new Dictionary<string, string>();

        for (int i = 0; i < n; i++) {
            string[] entry = Console.ReadLine().Split(' ');
            string name = entry[0];
            string number = entry[1];
            phoneBook[name] = number;
        }

        string query;
        while ((query = Console.ReadLine()) != null) {
            if (phoneBook.ContainsKey(query)) {
                Console.WriteLine($"{query}={phoneBook[query]}");
            } else {
                Console.WriteLine("Not found");
            }
        }
    }
}